### ISSUE TITLE
Add Relief

### ISSUE THREAD

**Issue by davelab6** [google/fonts#3271]
https://github.com/typomorpho/relief

A skeleton font that will need outlining.

**Comment by RosaWagner** [comment]
Personal note
The font in single line with open path. Make Glyphs3 crash, so need to work with Glyphs2. Simon is working on a ufostroker filter to export the font with close contours, and Matt on a rounded corner filter.

in venv:
- gftools
- pip install git+https://github.com/simoncozens/ufo2ft@pre-filter-string
- pip install ufo2ft[compreffor]>=2.20.0;2D
- pip install git+https://github.com/simoncozens/ufo2ft@pre-filter-string

then cmd is: `fontmake --filter 'ufostroker::StrokeFilter(Width=40,pre=True)' -g Relief.glyphs -o ttf`
Font build, but still corners issues, we would like a complete noodled outline:
<img width="1236" alt="Capture d’écran 2021-10-13 à 12 52 28" src="https://user-images.githubusercontent.com/12222436/137124441-8023d063-e977-49b1-b550-678f840aceb5.png">

**Comment by RosaWagner** [comment]
These are the problems so far that block us in the onbaording this project:

**Issue with diagonals joining with short segments:**
closed issues but unfortunately unsolved: https://github.com/MFEK/math.rlib/issues/16

<img width="824" alt="Capture d’écran 2022-04-14 à 09 49 33" src="https://user-images.githubusercontent.com/12222436/163339957-f24472ac-0920-4d68-afd4-45c4b4e121c9.png">
<img width="983" alt="Capture d’écran 2022-04-14 à 09 36 50" src="https://user-images.githubusercontent.com/12222436/163340004-cb6af9d0-1bfc-4b51-a717-4cb05fe45ddb.png">

**Issue with decomposed transformed components filters:**
open issue: https://github.com/simoncozens/ufostroker/issues/2

<img width="691" alt="Capture d’écran 2022-04-14 à 09 40 48" src="https://user-images.githubusercontent.com/12222436/163340032-38eaa1d3-08f0-468a-8f62-48ed725c1b0c.png">

using ufostroker 0.2.3

Note to @davelab6:
The idea for this project is to avoid multiple sources for multiple exports — even if the authors decided to make another repo for the font they actually published (https://github.com/isdat-type/Relief-SingleLine). But we are in a dead end for few months now, so their idea is to outline the source to make a releasable font for GF, that would be called "Relief" instead of "Relief SingleLine". This is not ideal solution but they are eager to move forward on the onboarding process since there is no movement on our end to improve the tooling.

**Comment by isdat-type** [comment]
From our own tests, the Expand Stroke function in FontForge seems to remain the best rendering from our skeletal / monolinear drawings, see screenshots (even slightly better than Noodler script on Glyphs App and similar to Outliner script in Robofont). Can Ufostroker use the script library behind Expand Stroke in FontForge to improve the TTF / Fontmake / MFEK export process? Apparently @ctrlcctrlv already mentioned similar remarks earlier.

![relief_fontforge_expandstroke_01](https://user-images.githubusercontent.com/82975340/163555626-ca7462a0-8447-4e63-a7a7-7a83572dd5e6.jpg)
![relief_fontforge_expandstroke_02](https://user-images.githubusercontent.com/82975340/163555638-e24cc16d-ad0f-49a7-ab6a-93f262185efa.jpg)
![relief_fontforge_expandstroke_03](https://user-images.githubusercontent.com/82975340/163555642-94714f7f-1a9b-4ca9-9397-8365fbfd3a9c.jpg)
![relief_fontforge_expandstroke_04](https://user-images.githubusercontent.com/82975340/163555648-3d19eae8-4be5-43b6-91ec-3c9a3c65c444.jpg)

**Comment by ctrlcctrlv** [comment]
I replied at https://github.com/MFEK/math.rlib/issues/16#issuecomment-1100855413.

I am only interested in _potentially_ bringing FontForge's stroke function into MFEKstroke to the extent that it makes nib stroking possible, if I don't end up solving that in pure Rust first.

**Comment by davelab6** [comment]
@simoncozens I wonder if this would be interesting for your builder2 work :)

**Comment by simoncozens** [comment]
If we have a good stroker, then this ties in a lot with [Dotter](https://github.com/simoncozens/Dotter); in fact the hard bit is already done as we are using stroking in builder2 in SIL projects.

The problem here is we're blocked on the current output of ufostroker being suboptimal for small path segments. Porting the fontforge expand stroke logic might move us forward, but that's a big job.

**Comment by emmamarichal** [comment]
I just run a quick QA on it, and this is the rendering I got for ttf. So I guess we would need more investigations on it!

<img width="1195" alt="Image" src="https://github.com/user-attachments/assets/8a7ded91-d8ca-431f-813f-2d64c51b5602" />
<img width="1471" alt="Image" src="https://github.com/user-attachments/assets/b14e23d8-f22a-4304-87ec-98ee0e5a497a" />
<img width="1484" alt="Image" src="https://github.com/user-attachments/assets/3c45960e-ab6f-4fdc-8399-d5d91e8c4e9f" />

**Comment by isdat-type** [comment]
@davelab6: A good stroker with round endings compatible with the Google Font TTF export process is still needed; we can nevertheless provide a source file containing an already stroked line as an alternative for a possible export. This solution had been rejected in 2021—22 because of the necessity to maintain 2 distinct source files. But still doable. 

@emmamarichal: The test feedback you have is absolutely normal because the TTF file we are proposing in the Relief SingleLine repository is strictly for CAD uses: it is an open-paths skeletal design automatically closed through the TTF export that is re-opened by some CAD softwares (as Rhinoceros 3D and others) to achieve a true single-stroke usage. Please check the [TTF part](https://github.com/isdat-type/Relief-SingleLine#cad-softwares-closed-ttf-towards-open-paths-method) of our manual in the Relief SingleLine Readme file or have a look to our Rhino 7 install and use [previews](https://github.com/isdat-type/Relief-SingleLine/tree/main/fonts/open_ttf/Rhino3D_Relief-SingleLine_in_use).

**Comment by emmamarichal** [comment]
@isdat-type thanks for the explanations! I went a bit fast and didn't take the time to dive deeper into the project, that now more clear :)

**Comment by simoncozens** [comment]
I seem to have created a path stroker which handles all the troublesome glyphs in this font.

<img width="830" height="383" alt="Image" src="https://github.com/user-attachments/assets/4419e3d5-e6e3-494b-a320-c866d70515aa" />
